src/rust/src/asn1.rs

```diff
@@ -95,11 +96,71 @@ fn parse_spki_for_data(py: pyo3::Python<'_>, data: &[u8]) -> pyo3::PyResult<pyo3
     Ok(pyo3::types::PyBytes::new(py, result).to_object(py))
 }
 
+fn big_asn1_uint_to_py<'p>(
+    py: pyo3::Python<'p>,
+    v: asn1::BigUint,
+) -> pyo3::PyResult<&'p pyo3::PyAny> {
+    let int_type = py.get_type::<pyo3::types::PyLong>();
+    int_type.call_method1("from_bytes", (v.as_bytes(), "big"))
+}
+
+#[pyo3::prelude::pyfunction]
+fn decode_dss_signature(py: pyo3::Python<'_>, data: &[u8]) -> pyo3::PyResult<pyo3::PyObject> {
+    let (r, s) = asn1::parse::<_, PyAsn1Error, _>(data, |p| {
+        p.read_element::<asn1::Sequence>()?.parse(|p| {
+            let r = p.read_element::<asn1::BigUint>()?;
+            let s = p.read_element::<asn1::BigUint>()?;
+            Ok((r, s))
+        })
+    })?;
+
+    Ok((big_asn1_uint_to_py(py, r)?, big_asn1_uint_to_py(py, s)?).to_object(py))
+}
+
+fn py_int_to_big_endian_bytes(v: &pyo3::types::PyLong) -> pyo3::PyResult<&[u8]> {
+    // Round the length up so that we prefix an extra \x00. This ensures that
+    // integers that'd have the high bit set in their first octet are not
+    // encoded as negative in DER.
+    let n = v.call_method0("bit_length")?.extract::<usize>()? / 8 + 1;
+    v.call_method1("to_bytes", (n, "big"))?.extract()
+}
+
+#[pyo3::prelude::pyfunction]
+fn encode_dss_signature(
+    py: pyo3::Python<'_>,
+    r: &pyo3::types::PyLong,
+    s: &pyo3::types::PyLong,
+) -> pyo3::PyResult<pyo3::PyObject> {
+    if r.rich_compare((0).to_object(py), CompareOp::Lt)?
```

Comment:
How do we ensure coverage when hoisting these types of branches into rust?

Reply:
Well, we could figure out Rust coverage in CI? Do you want that to be a pre-req to this? (Is this literally the first branch in the rust code? Neat!)

Reply:
`parse_spki_for_data` has a branch too, but it slipped my mind in that review 😄

Yes I think we should figure that out, although I'm a bit worried given that our hosted coverage service doesn't handle a single language well...

Reply:
Yeah, tooling here will be... fascinating. I'll investigate.